chore: add license

AlexD10S opened the pull request.

brunopgalvao reviewed (commented):
https://github.com/paritytech/polkadot-sdk/blob/master/substrate/HEADER-APACHE2
- Do we need a HEADER file?

brunopgalvao commented:
The legal team at Parity requested to add the following at the top of every file:
- https://github.com/paritytech/polkadot-sdk/blob/3410dfb3929462da88be2da813f121d8b1cf46b3/substrate/frame/assets/src/benchmarking.rs#L1-L16

brunopgalvao commented:
I think we also need a HEADER file. Examples:
- https://github.com/paritytech/ink/blob/master/FILE_HEADER
- https://github.com/paritytech/polkadot-sdk/blob/master/substrate/HEADER-APACHE2

brunopgalvao commented:
Worth mentioning that if we refactor to crates, there should be a LICENSE within the crate:
- https://github.com/paritytech/ink/tree/master/crates/e2e

AlexD10S commented:
> The legal team at Parity requested to add the following at the top of every file:
> 
> * https://github.com/paritytech/polkadot-sdk/blob/3410dfb3929462da88be2da813f121d8b1cf46b3/substrate/frame/assets/src/benchmarking.rs#L1-L16
> * https://github.com/paritytech/ink/blob/bd488de08bfaf2ee63d4600d3ae727c51e767032/crates/allocator/src/lib.rs#L1-L19

For me this seems excessive, but I have added it following the experts advice.
Interesting discussion here  https://opensource.stackexchange.com/questions/292/should-i-include-a-copyright-line-in-every-file

AlexD10S commented:
> Worth mentioning that if we refactor to crates, there should be a LICENSE within the crate:
> 
> * https://github.com/paritytech/ink/tree/master/crates/e2e

Yes! But it seems is only a matter of pointing to the other LICENSE file (If we are using the same)

brunopgalvao commented:
> > The legal team at Parity requested to add the following at the top of every file:
> 
> > 
> 
> > * https://github.com/paritytech/polkadot-sdk/blob/3410dfb3929462da88be2da813f121d8b1cf46b3/substrate/frame/assets/src/benchmarking.rs#L1-L16
> 
> > * https://github.com/paritytech/ink/blob/bd488de08bfaf2ee63d4600d3ae727c51e767032/crates/allocator/src/lib.rs#L1-L19
> 
> 
> 
> For me this seems excessive, but I have added it following the experts advice.
> 
> Interesting discussion here  https://opensource.stackexchange.com/questions/292/should-i-include-a-copyright-line-in-every-file

We could do what Linux does and add a one-liner to the top:
- https://github.com/torvalds/linux/blob/master/rust/macros/helpers.rs

AlexD10S commented:
> > > The legal team at Parity requested to add the following at the top of every file:
> > 
> > 
> > > 
> > 
> > 
> > > * https://github.com/paritytech/polkadot-sdk/blob/3410dfb3929462da88be2da813f121d8b1cf46b3/substrate/frame/assets/src/benchmarking.rs#L1-L16
> > 
> > 
> > > * https://github.com/paritytech/ink/blob/bd488de08bfaf2ee63d4600d3ae727c51e767032/crates/allocator/src/lib.rs#L1-L19
> > 
> > 
> > For me this seems excessive, but I have added it following the experts advice.
> > Interesting discussion here  https://opensource.stackexchange.com/questions/292/should-i-include-a-copyright-line-in-every-file
> 
> We could do what Linux does and add a one-liner to the top:
> 
> * https://github.com/torvalds/linux/blob/master/rust/macros/helpers.rs

I prefer the Linux approach too with only one line comment per file.  Updated

Daanvdplas reviewed (commented):
Do the files in `tests/` & `templates/` not require a header of the License? Other than that, thanks for taking this on Alex!

weezy20 commented on crates/pop-parachains/Cargo.toml:
```diff
@@ -3,6 +3,7 @@ name = "pop-parachains"
 description = "Library for generating, building and running parachains."
 version = "0.1.0"
 edition = "2021"
+license = "Apache-2.0"
```
`pop-parachains` and `pop-contract` are Apache 2.0

brunopgalvao replied:
> Copyleft vs. Permissive: GPL-3.0 is a copyleft license, which means that derivative works must also be licensed under the GPL-3.0 or a compatible license, and the source code must be made available to recipients. Apache 2.0 is a permissive license, which allows derivative works to be distributed under any license, including proprietary licenses, and does not require the distribution of source code.

The idea is that, with Apache 2.0, people can create products around.
Whereas with the code, devs can use it as long as they add the GPL license and make the code available.

brunopgalvao replied:
LGTM.
